tests/contest/contest/src/tests/linux_masked_paths/masked_paths.rs

```diff
@@ -0,0 +1,251 @@
+        // write something.
+        let test_sub_sub_file = bundle_path.join(&masked_file_sub_sub);
+        if let Err(e) = fs::File::create(&test_sub_sub_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_sub_sub_file, b"secrets") {
+            bail!(e)
+        }
+
+        let test_sub_file = bundle_path.join(&masked_file_sub);
+        if let Err(e) = fs::File::create(&test_sub_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_sub_file, b"secrets") {
+            bail!(e)
+        }
+
+        let test_file = bundle_path.join(masked_file);
+        if let Err(e) = fs::File::create(&test_file) {
+            bail!(e)
+        }
+        if let Err(e) = fs::write(&test_file, b"secrets") {
+            bail!(e)
+        }
+
+        Ok(())
+    })
+}
+
+fn check_masked_rel_paths() -> TestResult {
+    // Deliberately set a relative path to be masked,
+    // and expect an error
+    let masked_rel_path = "masked_rel_path";
+    let masked_paths = vec![masked_rel_path.to_string()];
+    let spec = get_spec(masked_paths);
+
+    test_inside_container(spec, &|bundle_path| {
+        use std::{fs, io};
+        let test_file = bundle_path.join(masked_rel_path);
+        match fs::metadata(&test_file) {
+            io::Result::Ok(md) => {
+                bail!(
+                    "reading path {:?} should have given error, found {:?} instead",
+                    test_file,
+                    md
+                )
+            }
+            io::Result::Err(e) => {
+                let err = e.kind();
+                if let io::ErrorKind::NotFound = err {
+                    Ok(())
+                } else {
+                    bail!("expected not found error, got {:?}", err);
+                }
+            }
+        }
+    })
+}
+
+fn check_masked_symlinks() -> TestResult {
+    // Deliberately create a masked symlink that points an invalid file,
+    // and expect an error.
+    let root = PathBuf::from("/");
+    let masked_symlink = "masked_symlink";
+    let masked_paths = vec![root.join(masked_symlink).to_string_lossy().to_string()];
+    let spec = get_spec(masked_paths);
+
+    let res = test_inside_container(spec, &|bundle_path| {
+        use std::{fs, io};
+        let test_file = bundle_path.join(masked_symlink);
+        // ln -s .. /masked-symlink ; readlink -f /masked-symlink; ls -L /masked-symlink
+        match std::os::unix::fs::symlink("../masked_symlink", &test_file) {
+            io::Result::Ok(_) => { /* This is expected */ }
+            io::Result::Err(e) => {
+                bail!("error in creating symlink, to {:?} {:?}", test_file, e);
+            }
+        }
+
+        let r_path = match fs::read_link(&test_file) {
+            io::Result::Ok(p) => p,
+            io::Result::Err(e) => {
+                bail!("error in reading symlink at {:?} : {:?}", test_file, e);
+            }
+        };
+
+        match fs::metadata(r_path) {
```

Comment:
same as one before.

Reply:
In this case, we are creating a symbolic link to a non-existent path and verifying that the symlink cannot be read as expected. If the relative path is not set correctly within the container's root filesystem, resolving the link will be problematic, leading to an error. This test aims to detect that scenario.

The validation for `fs::metadata` will be similar to the following comment.
https://github.com/youki-dev/youki/pull/2950#discussion_r1824747287